pymodbus/transport/transport.py

```diff
@@ -469,6 +471,12 @@ def handle_new_connection(self) -> ModbusProtocol:
 
     async def do_reconnect(self) -> None:
         """Handle reconnect as a task."""
+        if not (
```

Comment:
We try, in our internal functions, not to have
- exceptions due to bad calling.
- I believe we always call do_reconnect, and it then decides what actually to do.

Reply:
> We try, in our internal functions, not to have exceptions due to bad calling.

Yes, this is why I split all these `AssertionError` into a separate commit.

> I believe we always call do_reconnect, and it then decides what actually to do.

Unfortunately with the various conditionals it's not possible to prove this is always called correctly.  (`mypy` would not complain in that case).  If called incorrectly, it will raise an unhandled `ValueError`

Reply:
into a separate commit does not help, I do not review your individual commits, that would potentially mean adding review comments you have solved in a later commit.

Apart from that all commits in one PR is merged, so later it is all mixed.

Reply:
I know I can do that...but could we please keep 1 PR with the type changes that are really needed, which I am most likely to approve without a lot of comments, and a second PR with the code changes, where I am likely to have a lot more opinions.

I might be old-fashioned, but I see a PR as a unit, I have never cared about the single commits....the reason being that when I make comments the author add commits etc....and the only thing that counts is the sum of all the commits,  because that is what will be merged.